README.adoc

```diff
@@ -4294,6 +4294,45 @@ hash = { one: 1, two: 2, three: 3 }
 
 Avoid the use of mutable objects as hash keys.
 
+=== No Mutable Defaults [[no-mutable-defaults]]
+
+Avoid the use of shared mutable objects as hash default values.
+
+Creating a Hash in such a way will share the default value
+across all keys, causing unexpected behavior when modifying it.
+
+For example, when the Hash was created with an Array as the argument,
+calling `hash[:foo] << 'bar'` will also change the value of all
+other keys that have not been explicitly assigned to.
+
+[source,ruby]
+----
+# bad
+Hash.new([])
+Hash.new({})
+Hash.new(Array.new)
+Hash.new(Hash.new)
+
+# okay -- beware this will silently discard mutations and only remember assignments
```

Comment:
Are there any legitimate uses for this? Can we say 'probably bad' instead of 'okay'?